src/scripts/generator.ts

```diff
@@ -109,7 +109,13 @@ export class Generator {
   }
 
   private cleanPropertyName(prop: string) {
-    return prop.replace('@', '').replace('-', '');
+    const match = prop.match(/@|\-|\./g);
+    return match ? `'${prop}'` : prop;
+  }
+
+  private hasResourceParam(method: SchemaMethod) {
+    return method.parameters &&
+        Object.keys(method.parameters).indexOf('resource') > -1;
```

Comment:
why not just `'resource' in method.parameters`?

Reply:
Because I just love using `Object.keys` :) . Good catch.